Sources/SwiftSyntax/SyntaxCollections.swift.gyb

```diff
@@ -203,12 +203,32 @@ public struct ${node.name}: _SyntaxBase, Hashable, SyntaxCollection {
 }
 
 /// Conformance for `${node.name}`` to the Sequence protocol.
-extension ${node.name}: Sequence {
-  fileprivate static func nextElement<Iter>(
-    _ iterator: inout Iter, parent: _SyntaxBase
-  ) -> ${node.collection_element_type}? where Iter: AbsoluteRawSyntaxIteratorProtocol {
-    guard let absoluteRaw = iterator.next() else { return nil }
-    let data = SyntaxData(absoluteRaw, parent: parent)
+extension ${node.name}: BidirectionalCollection {
+  public typealias Element = ${node.collection_element_type}
+  public typealias Index = SyntaxChildrenIndex
+
+  private var presentRawChildren: PresentRawSyntaxChildren {
+    return PresentRawSyntaxChildren(self.data.absoluteRaw)
+  }
```

Comment:
`SyntaxCollection` types are essentially arrays of nodes, they should not be containing `nil` children (e.g. note that the implementation of `SyntaxCollection.count` is based on this invariant), we should use `RawSyntaxChildren` here since that one has more efficient index iteration.

They could also get a `distance(from:to:)` implementation.

Reply:
Good catch. Thanks!